middleware/logger.js

```diff
@@ -0,0 +1,56 @@
+const shortId = require('shortid');
+
+/**
+ * Logs requests
+ */
+module.exports = (req, res, next) => {
+  const cleanup = () => {
+    res.removeListener('finish', log);
+    res.removeListener('close', abort);
+    res.removeListener('error', error);
+  };
+
+  const log = () => {
+    cleanup();
+    const logger = getLoggerByStatusCode(res.statusCode);
+    logger(
+      `${getPrefix(req)} ${res.statusCode} ${res.statusMessage} ${res.get(
+        'Content-Length',
+      ) || 0}b sent`,
+    );
+  };
+
+  const abort = () => {
+    cleanup();
+    console.warn(`${getPrefix(req)} Request aborted by client`);
+  };
+
+  const error = error => {
+    cleanup();
+    console.error(`${getPrefix(req)} Request pipeline error: ${error}`);
+  };
+
+  req.requestId = shortId.generate();
+
+  console.info(`${getPrefix(req)} ${req.method} ${req.originalUrl}`);
+
+  res.on('finish', log);
+  res.on('close', abort);
+  res.on('error', error);
+
+  next();
+};
+
```

Comment:
As usual, JDoc comment pls
```suggestion
/**
 * Does some stuff
 * @param {??} req 
 * @returns {??} some stuff
 */
```